api/openapi.yaml

```diff
@@ -0,0 +1,393 @@
+        '500':
+          description: Internal server error
+    post:
+      tags:
+      - Users
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+      responses:
+        '201':
+          description: Successfully created user
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        '400':
+          description: Invalid User object
+        '500':
+          description: Internal server error
+  /users/{id}:
+    parameters:
+    - name: id
+      in: path
+      schema:
+        type: integer
+      required: true
+    get:
+      tags:
+      - Users
+      responses:
+        '200':
+          description: Successfully retrieved user
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        '404':
+          description: User with this id not found
+        '500':
+          description: Internal server error
+    put:
+      tags:
+      - Users
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+      responses:
+        '200':
+          description: Successfully updated user
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        '400':
+          description: Invalid User object
+        '500':
+          description: Internal server error
+    delete:
+      tags:
+      - Users
+      responses:
+        '204':
+          description: User successfully deleted
+        '404':
+          description: User with this id not found
+        '500':
+          description: Internal server error
+  /users/{id}/followers:
+    parameters:
+    - name: id
+      in: path
+      schema:
+        type: integer
+      required: true
+    get:
+      tags:
+      - Users
+      responses:
+        '200':
+          description: Follower users successfully retrieved
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/User'
+        '500':
+          description: Internal server error
+  /users/{id}/following:
+    parameters:
+    - name: id
+      in: path
+      schema:
+        type: integer
+      required: true
+    get:
+      tags:
+      - Users
+      responses:
+        '200':
+          description: Following users successfully retrieved
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/User'
+        '500':
+          description: Internal server error
+  /messages:
+    get:
+      tags:
+      - Messages
+      parameters:
+      - name: poster
```

Comment:
Is this a parameter passed in the request body?

Reply:
Correct, it would look something like `http://localhost:5000/messages?poster=2`. Something important that's come up in the past when working with apis is that here, `poster` is a unique id of a user. So querying using this parameter should only ever return 1 Message or 0 Messages. This is in our GET /messages endpoint though which is meant to return an array of messages. This is perfectly fine to do, we just need to remember that the format will be a singleton array or an empty array when we go to parse it.

Reply:
What about a user who has posted multiple messages?

Reply:
Oh right, I was getting mixed up haha.

Reply:
Gotcha, so it returns each message posted by the user?

Reply:
Right

Reply:
Looks good then, thanks for the clarification